mods/ra/maps/soviet-06b/soviet06b-AI.lua

```diff
@@ -0,0 +1,112 @@
+IdleHunt = function(unit) if not unit.IsDead then Trigger.OnIdle(unit, unit.Hunt) end end
+
+IdlingUnits = function()
+	local lazyUnits = enemy.GetGroundAttackers()
+
+	Utils.Do(lazyUnits, function(unit)
+		Trigger.OnDamaged(unit, function()
+			Trigger.ClearAll(unit)
+			Trigger.AfterDelay(0, function() IdleHunt(unit) end)
+		end)
+	end)
+end
+
+BaseBuildings =
+{
+	{ "apwr", CVec.New(5, -9), 500, true },
+	{ "tent", CVec.New(-4, -4), 400, true },
+	{ "proc", CVec.New(0, -8), 1400, true },
+	{ "weap", CVec.New(-4, -8), 2000, true },
+	{ "apwr", CVec.New(6, -5), 500, true }
+}
+
+BuildBase = function()
+	if CYard.IsDead or CYard.Owner ~= enemy then
+		return
+	elseif Harvester.IsDead and enemy.Resources <= 299 then
+		return
+	end
+
+	for i,v in ipairs(BaseBuildings) do
+		if not v[4] then
+			BuildBuilding(v)
+			return
+		end
+	end
+
+	Trigger.AfterDelay(DateTime.Seconds(10), BuildBase)
+end
+
+BuildBuilding = function(building)
+	Trigger.AfterDelay(Actor.BuildTime(building[1]), function()
+		local actor = Actor.Create(building[1], true, { Owner = enemy, Location = CYardLocation.Location + building[2] })
+		enemy.Cash = enemy.Cash - building[3]
+
+		building[4] = true
+		Trigger.OnKilled(actor, function() building[4] = true end)
```

Comment:
If not using tables, then it should be `Trigger.OnKilled(actor, function() building[4] = false end)` like the triggers below.